File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
         int left = 0; // Pointer to place the next non-zero element

    // Traverse the array with right pointer
    for (int right = 0; right < nums.size(); ++right) {
        if (nums[right] != 0) {
            // Swap non-zero element to the left pointer's position
            std::swap(nums[left], nums[right]);
            left++;
        }
    }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, arr: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        count=0
        i=0
        while i<len(arr):
            if arr[i]==0:
                count+=1
                arr.pop(i)
            else:
                i+=1
        arr.extend([0]*count)
        